compiler/passes/codegen.cpp

```diff
@@ -530,6 +530,67 @@ static void codegen_header_compilation_config() {
 }
 
 
+static void protectNameFromC(Symbol* sym) {
+  //
+  // Let's assume we only have to rename user symbols and that we've
+  // done a good job of protecting our internal and standard module
+  // symbols.  This has the advantage of not having to take special
+  // pains to keep from renaming things like chpl_string and uint64_t
+  // (which perhaps should arguably also have FLAG_EXTERN, but don't
+  // consistently seem to today; and adding FLAG_EXTERN to them seems
+  // to result in it getting propagated to type aliases and such in
+  // ways that caused headaches for me).
+  //
```

Comment:
This seems a little unfortunate. I think long term we probably want to munge symbols from internal/standard modules. I think we can assume 'public' functions in the internal modules are safe (or will be long term if they start with chpl_) but I'm sure we have helper routines and configs that could just as easily conflict with a library as user code. 

I don't think this commit should be put on hold, just that it would be nice to do in the future. 

What are dists considered? Do they fall in into the standard module category?